### ISSUE TITLE
Filter Nested relationship unknown column , Laravel Prefix Table problem

### ISSUE THREAD

**Issue by kouloughli-hemza** [yajra/laravel-datatables#2024]
### Prefix problem not yet fixed in relationship
i created custom dropdown to filter using aganecy name;
in global search it was working but when i added custom filter i get the following error
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'table.name' in 'where clause'
in matter of sql it expected error because Yajra not adding PREFIX to this table,
not sure why , only in where clause the table not getting prefixed
```php            
$data =  $model->with([
                'Table',
                'Table1.relation_1',
                'Table1.relation_2',
                'Table1.relation_3'
            ])->orderBy('table0.id','desc')->select();
            return $data;
```

This is the generated Sql query : 

```sql
select * from `prefix_table0` left join `prefix_table1` on `prefix_table0`.`id_application` = `prefix_table1`.`id` left join `prefix_table2` on `prefix_table1`.`created_by_agency` = `prefix_tbale2`.`id` where `prefix_table1`.`deleted_at` is null and LOWER(table2.name) REGEXP 88 order by `dz_table0`.`id` desc
```
as you can see the table2 is not getting prefixed ,
this is commun problem in yajra 
im not sure if there is a solution to this

### System details

- Operating System: ubuntu
- PHP Version : 7.2
- Laravel Version : Laravel Framework 5.6.11

**Comment by yajra** [comment]
It seems like you are using a self join relationship and it is currently not supported by the package. You need to use join statements for this to work on self joins.

**Comment by yajra** [comment]
Fixed via #2051.
